File: problems_026to050/problem_032/solution.py
import os
import sys
from math import factorial

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import nth_lexicographic_permutation


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None

        pandigital_products = set()

        for order in range(1, factorial(9)+1):
            permutation = nth_lexicographic_permutation(order, list('123456789'))
            product = int(permutation[5:])

            if int(permutation[0:1]) * int(permutation[1:5]) == product or int(permutation[0:2]) * int(permutation[2:5]) == product:
                pandigital_products.add(product)

        answer = sum(pandigital_products)

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_032/solution.ts
import { SolutionFunction } from '../../lib/types';
import { sum, factorial, nthLexicographicPermutation } from '../../lib/toolbox';

const solution: SolutionFunction = () => {
  let answer: ReturnType<SolutionFunction> = 0;

  const pandigitalProducts: Set<number> = new Set();
  const limit = Number(factorial(9));

  for (let order = 1; order <= limit; order += 1) {
    const permutation = nthLexicographicPermutation(order, '123456789'.split(''));
    const product = Number(permutation.slice(5));

    if (
      Number(permutation.slice(0, 1)) * Number(permutation.slice(1, 5)) === product
        || Number(permutation.slice(0, 2)) * Number(permutation.slice(2, 5)) === product
    ) {
      pandigitalProducts.add(product);
    }
  }

  answer = sum([...pandigitalProducts]);

  return answer;
};

export default solution;
